flake-modules/dev/default.nix

```diff
@@ -18,6 +18,7 @@
     lib.optionalAttrs (inputs.treefmt-nix ? flakeModule) {
       treefmt.config = {
         projectRootFile = "flake.nix";
+        flakeCheck = true;
```

Comment:
What is the difference between this check and the pre-commit check?

Reply:
This one adds a flake check that checks `treefmt` passes. It's actually true by default, I just made it explicit here. See [`perSystem.treefmt.flakeCheck`](https://flake.parts/options/treefmt-nix#opt-perSystem.treefmt.flakeCheck).

The `pre-commit` check adds a check that `pre-commit run --all-files` passes. See [`perSystem.pre-commit.check.enable`](https://flake.parts/options/pre-commit-hooks-nix#opt-perSystem.pre-commit.check.enable).

As with all derivation builders (and therefor flake checks), they are run within the nix sanbox. Because I've added a pre-commit hook that _itself_ does a nix build, that will always fail because we're already in the nix sandbox when running it in the check.